packages/zoe/src/contracts/attestation/types.js

```diff
@@ -0,0 +1,182 @@
+ * Given an address string and an Amount to add a lien on and the
+ * expiration, record that the amount is "liened", and mint an
+ * attestation payment for the amount. Return the payment.
+ *
+ * @param {Address} address
+ * @param {Amount} amount
+ * @param {Timestamp} expiration
+ * @returns {Promise<Payment>}
+ */
+
+/**
+ * @callback AddReturnableLien
+ *
+ * Given an address string and an Amount to put a lien on, record that
+ * the Amount is "liened", and mint an attestation payment for the
+ * amount liened. Return the payment.
+ *
+ * @param {Address} address
+ * @param {Amount} amount
+ * @returns {Promise<Payment>}
+ */
+
+/**
+ * @typedef {Object} ExpiringAttElem
+ * @property {Address} address
+ * @property {Amount} amountLiened
+ * @property {Timestamp} expiration
+ * @property {Handle<'attestation'>} handle
+ */
+
+/**
+ * @typedef {Object} ReturnableAttElem
+ * @property {Address} address
+ * @property {Amount} amountLiened
+ */
+
+/**
+ * @callback ReturnAttestation
+ * @param {ZCFSeat} seat
+ * @returns {void}
+ */
+
+/**
+ * @callback DisallowExtensions
+ * @param {Address} address
+ * @returns {void}
+ */
+
+/**
+ * @callback ExtendExpiration
+ *
+ * Scenario: I have a vote ongoing where I've escrowed a payment and
+ * gotten it back. I would like to participate in a second vote with
+ * the election closing after the expiration of my payment. I want to
+ * extend my payment's expiration to be after the second vote's
+ * closing. I must not be able to trick the first vote into escrowing
+ * my new payment and double-counting my influence.
+ *
+ * Only the owner of the liened tokens is allowed to extend the
+ * expiration. We will use the holder of the original attestation
+ * payment as the measure of this. We can only extend the expiration
+ * if we have not already dropped the lien. Therefore to extend, we
+ * must still have a record of the attestation (with the unique
+ * handle).
+ *
+ * @param {ZCFSeat} seat
+ * @param {Timestamp} newExpiration
+ * @returns {void}
+ */
+
+/**
+ * @callback Slashed
+ *
+ * On slashing, we disallow any extensions. We do not reduce the liens.
+ *
+ * @param {Address} address
+ * @param {Timestamp} currentTime
+ * @returns {void}
+ */
+
+/**
+ * @callback MakeReturnAttInvitation
+ *
+ * Make an invitation for returning a returnable attestation.
+ *
+ * @returns {Promise<Invitation>}
+ */
+
+/**
+ * @callback MakeExtendAttInvitationInternal
+ *
+ * Make an invitation for extending the expiration date of an expiring
+ * attestation. Internal because the currentTime must be passed in,
+ * and that can only come from a trusted source.
+ *
+ * @param {Timestamp} newExpiration
+ * @param {Timestamp} currentTime
+ * @returns {Promise<Invitation>}
+ */
+
+/**
+ * @callback MakeExtendAttInvitation
+ *
+ * Make an invitation for extending the expiration date of an expiring
+ * attestation.
+ *
+ * @param {Timestamp} newExpiration
+ * @returns {Promise<Invitation>}
+ */
+
+/**
+ * @callback GetLiened
```

Comment:
As I mentioned in chat, I'll only call this with one address at a time. Alternatively, I could call `GetLienedAmount` instead. Either way, I'll be passing in a `Brand` for the denom.

Reply:
Ah, thank you for correcting that. I was relying on memory and was convinced that it was the `slashed` call instead that would be called only one address at a time. I'll also take a Brand as an argument.